include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -226,14 +226,14 @@ struct cartesian_segments
             // division.
             BOOST_GEOMETRY_ASSERT(ratio.denominator() != 0);
 
-            typedef typename promote_integral<CoordinateType>::type promoted_type;
-
-            promoted_type const numerator
-                = boost::numeric_cast<promoted_type>(ratio.numerator());
-            promoted_type const denominator
-                = boost::numeric_cast<promoted_type>(ratio.denominator());
-            promoted_type const dx_promoted = boost::numeric_cast<promoted_type>(dx);
-            promoted_type const dy_promoted = boost::numeric_cast<promoted_type>(dy);
+            typedef typename promote_integral<CoordinateType>::type integral_type;
```

Comment:
This type may not be integral type. The purpose of this util is to promote only integral types to bigger integral types and leave other types as they were.

See: https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/promote_integral.hpp#L312

Reply:
Thanks I'll rename it. The point was that promoted_type shadows the outer type with the same name.
I can call it `bigger_type` for example

Reply:
Done